docs/nd4j/templates/quickstart.md

```diff
@@ -0,0 +1,381 @@
+
+<!--- TODO: Create HelloNd4J or Quickstart-nd4j repo in Deeplearning4J. -->
+```shell
+git clone https://github.com/RobAltena/HelloNd4J.git
+
+cd HelloNd4J
+
+mvn install
+
+mvn exec:java -Dexec.mainClass="HelloNd4j"
+```
+
+When everything is set up correctly you should see the following output:
+
+```shell
+SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
+SLF4J: Defaulting to no-operation (NOP) logger implementation
+SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
+[         0,         0]
+```
+
+## Basics
+<!--- TODO: We will put some into this page. Start with refering to existing doc. -->
+While this quickstart is being build, please refer to our existing 
+[basics usage](./nd4j-basics) document.
+
+The main feature of Nd4j is the versatile n-dimensional array interface called INDArray. To improve performance Nd4j uses off-heap memory to store data. The INDArray is different from standard Java arrays.
+
+Some of the key properties and methods for an INDArray x are as follows:
+
+```java
+// The number of axes (dimensions) of the array.
+int dimensions = x.shape().length;
+
+// The dimensions of the array. The size in each dimension.
+long[] shape = x.shape();
+
+// The total number of elements.
+length = x.length();
+
+// The type of the array elements. 
+DataType dt = x.dataType();
+```
+<!--- staying away from itemsize and data buffer. The numpy quickstart has these. -->
+
+### Array Creation
+To create INDArrays you use the static factory methods of the [Nd4j](https://deeplearning4j.org/api/latest/org/nd4j/linalg/factory/Nd4j.html ) class.
+
+<!--- We have good docs on creating INDArrays already.  -->
+<!--- https://deeplearning4j.org/docs/latest/nd4j-overview#creating -->
+The `Nd4j.create` function is overloaded to make it easy to create INDArrays from regular Java arrays. The example below uses Java `double` arrays. Similar create methods are overloaded for `float`, `int` and `long`.
+
+```java
+double arr_2d[][]={{1.0,2.0,3.0},{4.0,5.0,6.0},{7.0,8.0,9.0}};
+INDArray x_2d = Nd4j.create(arr_2d);
+
+double arr_1d[]={1.0,2.0,3.0};
+INDArray  x_1d = Nd4j.create(arr_1d);
+```
+
+There are overloaded `create` functions for INDArrays up to 4 dimensions. To create INDArrays with any shape use the `create` functions that take a shape as one of their arguments.
+
+```java
+double[] flat = ArrayUtil.flattenDoubleArray(myDoubleArray);
+int[] shape = ...;	//Array shape here
+INDArray myArr = Nd4j.create(flat,shape,'c');
+```
+
+Nd4j can create arrays initialized with zeros and ones using the functions `zeros` and `ones`. The `rand` function allows you to create an array initialized with random values.
+The default datatype of the INDarray created is `float`. Some overloads allow you to set the datatype.
+
+```java
+INDArray  x = Nd4j.zeros(5);
+//[         0,         0,         0,         0,         0], FLOAT
+
+int [] shape = {5};
+x = Nd4j.zeros(shape, DataType.DOUBLE);
```

Comment:
Use `Nd4j.zeros(DataType.DOUBLE, 5)`,  no need to do it in 2 lines with array for shape.
Maybe also show the import for DataType the first time it's used.